### PR TITLE
feat: Add e2e test for guardian directory search

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-2474

## Related PRs
https://github.com/USSF-ORBIT/ussf-portal-client/pull/1183
https://github.com/USSF-ORBIT/ussf-personnel-api/pull/33

### CODE REVIEW

File: e2e/playwright/feature/guardian-directory.spec.ts [hunk 1]
```diff
@@ -19,10 +19,7 @@ const test = base.extend<TestingLibraryFixtures & CustomFixtures>({
 
 const { expect } = test
 
-test('can view the guardian directory', async ({
-  page,
-  loginPage,
-}) => {
+test('can view the guardian directory', async ({ page, loginPage }) => {
```
Comment (abbyoung) [inline e2e/playwright/feature/guardian-directory.spec.ts:22]: i'm not sure why all the formatting changed on this file, maybe something different in my local prettier setup? maybe we need to set something up in the repo? sorry!

File: e2e/playwright/feature/guardian-directory.spec.ts [hunk 2]
```diff
@@ -31,62 +28,143 @@ test('can view the guardian directory', async ({
...
+    await page
+      .locator('[data-testid="guardian-directory-table"] tbody tr')
+      .count()
+  ).toBe(1)
+
+  // Searching for empty string should return all results
+  await page.getByTestId('textInput').click()
+  await page.getByTestId('textInput').fill('')
+  await page.getByTestId('form').getByTestId('button').click()
+  // Inexplicably, counting the rows does not work here,
```
Comment (abbyoung) [inline e2e/playwright/feature/guardian-directory.spec.ts:153]: This was the weirdest thing that stumped me EOD Friday, so I found another approach. For some reason, the code that worked on line 143, to get the rows and count them, will only return 1 here, even though running the test in headed mode shows all the rows displayed. Why!

File: e2e/playwright/feature/guardian-directory.spec.ts (hunk 1 above)
Comment (gidjin) [reply e2e/playwright/feature/guardian-directory.spec.ts:22]: Or the defaults of one of our editors is different. I've seen this type of thing in our yaml files too. Maybe we should make sure that prettier is setup in this repo the same as our other ones.

File: e2e/playwright/feature/guardian-directory.spec.ts (hunk 2 above)
Comment (gidjin) [reply e2e/playwright/feature/guardian-directory.spec.ts:153]: I wonder if it's some sort of browser timing issue. I'm good with how you worked around it.

**Review by gidjin — APPROVED** [review]

**Pull request merged**